src/options/mkoptions.py

```diff
@@ -226,6 +228,64 @@
   return Options::current()->wasSetByUser(*this);
 }}"""
 
+# Mode templates
+TPL_DECL_MODE_ENUM = \
+"""
+enum class {type}
+{{
+  {values}
+}};"""
+
+TPL_DECL_MODE_FUNC = \
+"""
+std::ostream&
+operator<<(std::ostream& out, {type} mode) CVC4_PUBLIC;"""
+
+TPL_IMPL_MODE_FUNC = TPL_DECL_MODE_FUNC[:-len(" CVC4_PUBLIC;")] + \
+"""
+{{
+  out << "{type}::";
+  switch(mode) {{{cases}
+    default:
+        Unreachable();
+  }}
+  return out;
+}}
+"""
+
+TPL_IMPL_MODE_CASE = \
+"""
+    case {type}::{enum}:
+      out << "{enum}";
+      break;"""
+
+TPL_DECL_MODE_HANDLER = \
+"""
+{type}
+stringTo{type}(std::string option, std::string optarg);"""
```

Comment:
Nit: Can't we make the `std::string` arguments `const std::string&` arguments here?